src/caja-information-panel.c

```diff
@@ -960,6 +960,8 @@ add_buttons_from_metadata (CajaInformationPanel *information_panel, const char *
                     command_string = g_strdup (temp_str + 1);
                     g_free (button_name);
 
+                    button_name = NULL;
```

Comment:
Just add `g_free (button_name);` after `for (index = 0; (term = terms[index]) != NULL; index++)`, before `g_strfreev (terms);`
https://github.com/mate-desktop/caja/blob/e6421559dd1c4431502551d665c90d44f08e40ce/src/caja-information-panel.c#L980

Reply:
confirmed, the warning in Clang with your suggested change is gone

other question, we have into the `loop for` the line:

`button_name = g_strdup (temp_str + 1);`

and `g_strdup` needs to be freed, so, we need to add before this line `if (button_name) g_free (button_name);` ?

If `g_strdup` is called several times without freed we have memory leak

Reply:
I'm agree, free **button_name** before new memory allocation, and after the loop end.

You can move `g_free (button_name);` from line
https://github.com/mate-desktop/caja/blob/e6421559dd1c4431502551d665c90d44f08e40ce/src/caja-information-panel.c#L961
to line
https://github.com/mate-desktop/caja/blob/e6421559dd1c4431502551d665c90d44f08e40ce/src/caja-information-panel.c#L948
And add `g_free (button_name);` before the `g_strfreev (terms);`

For loop: button iteration followed by script iterarion -> proced to create button
For loop: button iteration followed by button iterarion -> can't create button without the script

Reply:
@rbuj  I did changes, please review again